### ISSUE TITLE
Anomaly detection models (vision)

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#532]
**Short Description**

![164736543-0ef58c66-dfb0-47c5-8e6e-26774dbc6fd3](https://user-images.githubusercontent.com/17668390/175886639-69a3524a-5784-4d59-ad72-2a75a1df0acb.gif)

Anomaly detection is an important category of common problem mostly in manufacturing industry. There is a good amount of ML models (and active research) to automate the anomaly detection task. I think it would be good addition to add these models in `keras_cv.models`. 

**Papers**

- [CutPaste](https://arxiv.org/abs/2104.04015) by Google.
- [FastFlow](https://arxiv.org/abs/2111.07677)
- [PatchCore](https://arxiv.org/abs/2106.08265)
- [PaDiM](https://link.springer.com/chapter/10.1007/978-3-030-68799-1_35)

Benchmark dataset

- [MVTec](https://www.mvtec.com/company/research/datasets/mvtec-ad)
- [BTech](https://paperswithcode.com/dataset/btad)

Metrics

- Image and Pixel-level AUC
- Image and Pixel-level F1 score

**Existing Implementations**

Currently, there are not much **strong support** available for anomaly detection (vision) in tensorflow.keras.  In contrast, a pytorch-lighting based implementations, named [anomalib](https://github.com/openvinotoolkit/anomalib) are available, maintained by a OpenVINO team. Currently it offers 10 different anomaly detection (vision) models and actively developing. These model are able to run in supervised (with binary mask) or unsupervised manner (with no mask). 

**Other Information**

If these models are welcomed, we can close this issue and create new ones for these models each.

**Comment by bhack** [comment]
Some of this model are available in a TF/Keras version at https://github.com/Burf/tfdetection

/cc @burf If he Is interested to contribute.

**Comment by bhack** [comment]
Is MVTec saturated?
https://paperswithcode.com/sota/anomaly-detection-on-mvtec-ad

**Comment by innat** [comment]
Yeah, looks so. Models like patch-core, or fast-flow (transformer based) left less room for improvement on this dataset.

**Comment by LukeWood** [comment]
At this time we're focusing on ironing out issues in our current offerings before expanding too broadly, lets revisit anomaly detection in the future when the more standard workflows are solved.
